{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants = pd.read_table(\"/Users/himanshujoshi/Desktop/LOVD variants/all_LOVD_variants.tsv.gz\", sep=\"\\t\")\n",
    "refseq_tx = pd.read_table(\"refseq_tx_chromosome_map.txt.gz\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prep Refseq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    35925\n",
       "True        25\n",
       "dtype: int64"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.any(refseq_tx.isna(), axis=1).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "refseq_tx.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    35925\n",
       "dtype: int64"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.any(refseq_tx.isna(), axis=1).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>RefSeq mRNA ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_152499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_173638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_001170755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_024097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>NM_001242750</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome/scaffold name RefSeq mRNA ID\n",
       "1                        1      NM_152499\n",
       "2                        1      NM_173638\n",
       "3                        1   NM_001170755\n",
       "4                        1      NM_024097\n",
       "5                        1   NM_001242750"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq_tx.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prep Variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Infer variant type (SNP or other) using regex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_pattern = re.compile(\"g.(\\d+)([ATCG])>([ATCG])\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants['Variant_Type'] = lovd_variants.DNA_change_genomic_hg19.apply(lambda x: 'SNP' if snp_pattern.match(x) is not None else 'Other')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SNP      260251\n",
       "Other     61750\n",
       "Name: Variant_Type, dtype: int64"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants.Variant_Type.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Obtain chromosome based on mRNA to Chr mapping (downloaded from GRCh37 Ensembl biomart)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants['RefSeq mRNA ID'] = lovd_variants.Transcript.str.split('.', expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants = lovd_variants.merge(refseq_tx, how='left', on='RefSeq mRNA ID')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get chr/pos/ref/alt for SNP variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_pos = lovd_variants.loc[lovd_variants['Variant_Type']=='SNP']['DNA_change_genomic_hg19'].apply(lambda x: pd.Series({'pos': snp_pattern.match(x)[1],\n",
    "                                                                                                              'ref': snp_pattern.match(x)[2],\n",
    "                                                                                                              'alt': snp_pattern.match(x)[3]}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants = lovd_variants.join(snp_pos)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a hgvs_g column so that we can use it to obtain VCF coordinates for non-SNP variants (Keep in mind, some of these will be invalid entries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DB_ID', 'Gene', 'Transcript', 'DNA_change_cDNA',\n",
       "       'DNA_change_genomic_hg19', 'Reported', 'Clinical_classification',\n",
       "       'Variant_remarks', 'Reference', 'Variant_Type', 'RefSeq mRNA ID',\n",
       "       'Chromosome/scaffold name', 'pos', 'ref', 'alt'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants['hgvs_g'] = lovd_variants.apply(lambda x: \"{}:{}\".format(x['Chromosome/scaffold name'], x['DNA_change_genomic_hg19']), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DB_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>DNA_change_cDNA</th>\n",
       "      <th>DNA_change_genomic_hg19</th>\n",
       "      <th>Reported</th>\n",
       "      <th>Clinical_classification</th>\n",
       "      <th>Variant_remarks</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>RefSeq mRNA ID</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>hgvs_g</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIF21A_000001</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.4497T&gt;C</td>\n",
       "      <td>g.39698647A&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39698647</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>12:g.39698647A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIF21A_000002</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3920G&gt;A</td>\n",
       "      <td>g.39709031C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39709031</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>12:g.39709031C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIF21A_000003</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3909-15G&gt;T</td>\n",
       "      <td>g.39709057C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39709057</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>12:g.39709057C&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIF21A_000004</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.1841G&gt;T</td>\n",
       "      <td>g.39735348C&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland, 174 he...</td>\n",
       "      <td>Faruq 2020, submtted</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39735348</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>12:g.39735348C&gt;A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           DB_ID    Gene   Transcript DNA_change_cDNA DNA_change_genomic_hg19  \\\n",
       "0  KIF21A_000001  KIF21A  NM_017641.3       c.4497T>C           g.39698647A>G   \n",
       "1  KIF21A_000002  KIF21A  NM_017641.3       c.3920G>A           g.39709031C>T   \n",
       "2  KIF21A_000003  KIF21A  NM_017641.3    c.3909-15G>T           g.39709057C>A   \n",
       "3  KIF21A_000004  KIF21A  NM_017641.3       c.1841G>T           g.39735348C>A   \n",
       "\n",
       "   Reported Clinical_classification  \\\n",
       "0         1                  benign   \n",
       "1         1           likely benign   \n",
       "2         1           likely benign   \n",
       "3         3                  benign   \n",
       "\n",
       "                                     Variant_remarks             Reference  \\\n",
       "0             VKGL data sharing initiative Nederland                     -   \n",
       "1             VKGL data sharing initiative Nederland                     -   \n",
       "2             VKGL data sharing initiative Nederland                     -   \n",
       "3  VKGL data sharing initiative Nederland, 174 he...  Faruq 2020, submtted   \n",
       "\n",
       "  Variant_Type RefSeq mRNA ID Chromosome/scaffold name       pos ref alt  \\\n",
       "0          SNP      NM_017641                       12  39698647   A   G   \n",
       "1          SNP      NM_017641                       12  39709031   C   T   \n",
       "2          SNP      NM_017641                       12  39709057   C   A   \n",
       "3          SNP      NM_017641                       12  39735348   C   A   \n",
       "\n",
       "             hgvs_g  \n",
       "0  12:g.39698647A>G  \n",
       "1  12:g.39709031C>T  \n",
       "2  12:g.39709057C>A  \n",
       "3  12:g.39735348C>A  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants.head(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants.query('Variant_Type != \"SNP\"').hgvs_g.to_csv(\"/Users/himanshujoshi/Desktop/LOVD variants/lovd_indels.csv\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once the indels have been run through VEP, download the VCF. The bits we're interested in can be obtained as follows:\n",
    "\n",
    "awk -F \"\\t\" 'NR >= 4 { print $3,$1,$2,$4,$5 }' OFS=\"\\t\" lovd_indels.vcf > lovd_indels.vcf.subset.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import VCF subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels_vcf = pd.read_table(\"/Users/himanshujoshi/Desktop/LOVD variants/lovd_indels.vcf.subset.tsv\", sep=\"\\t\", header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(50075, 5)"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indels_vcf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels_vcf.drop_duplicates(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(44892, 5)"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indels_vcf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [],
   "source": [
    "indels_vcf.rename(columns={'ID':'hgvs_g',\n",
    "                          '#CHROM':'vcf_chrom',\n",
    "                          'POS': 'vcf_pos',\n",
    "                          'REF': 'vcf_ref',\n",
    "                          'ALT': 'vcf_alt'}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine indels with lovd_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants2 = lovd_variants.merge(indels_vcf, on='hgvs_g', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DB_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>DNA_change_cDNA</th>\n",
       "      <th>DNA_change_genomic_hg19</th>\n",
       "      <th>Reported</th>\n",
       "      <th>Clinical_classification</th>\n",
       "      <th>Variant_remarks</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>RefSeq mRNA ID</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>hgvs_g</th>\n",
       "      <th>vcf_chrom</th>\n",
       "      <th>vcf_pos</th>\n",
       "      <th>vcf_ref</th>\n",
       "      <th>vcf_alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIF21A_000001</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.4497T&gt;C</td>\n",
       "      <td>g.39698647A&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39698647</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>12:g.39698647A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIF21A_000002</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3920G&gt;A</td>\n",
       "      <td>g.39709031C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39709031</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>12:g.39709031C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIF21A_000003</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3909-15G&gt;T</td>\n",
       "      <td>g.39709057C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39709057</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>12:g.39709057C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIF21A_000004</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.1841G&gt;T</td>\n",
       "      <td>g.39735348C&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland, 174 he...</td>\n",
       "      <td>Faruq 2020, submtted</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39735348</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>12:g.39735348C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIF21A_000005</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.1674-11C&gt;G</td>\n",
       "      <td>g.39735937G&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>NM_017641</td>\n",
       "      <td>12</td>\n",
       "      <td>39735937</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>12:g.39735937G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           DB_ID    Gene   Transcript DNA_change_cDNA DNA_change_genomic_hg19  \\\n",
       "0  KIF21A_000001  KIF21A  NM_017641.3       c.4497T>C           g.39698647A>G   \n",
       "1  KIF21A_000002  KIF21A  NM_017641.3       c.3920G>A           g.39709031C>T   \n",
       "2  KIF21A_000003  KIF21A  NM_017641.3    c.3909-15G>T           g.39709057C>A   \n",
       "3  KIF21A_000004  KIF21A  NM_017641.3       c.1841G>T           g.39735348C>A   \n",
       "4  KIF21A_000005  KIF21A  NM_017641.3    c.1674-11C>G           g.39735937G>C   \n",
       "\n",
       "   Reported Clinical_classification  \\\n",
       "0         1                  benign   \n",
       "1         1           likely benign   \n",
       "2         1           likely benign   \n",
       "3         3                  benign   \n",
       "4         1           likely benign   \n",
       "\n",
       "                                     Variant_remarks             Reference  \\\n",
       "0             VKGL data sharing initiative Nederland                     -   \n",
       "1             VKGL data sharing initiative Nederland                     -   \n",
       "2             VKGL data sharing initiative Nederland                     -   \n",
       "3  VKGL data sharing initiative Nederland, 174 he...  Faruq 2020, submtted   \n",
       "4             VKGL data sharing initiative Nederland                     -   \n",
       "\n",
       "  Variant_Type RefSeq mRNA ID Chromosome/scaffold name       pos ref alt  \\\n",
       "0          SNP      NM_017641                       12  39698647   A   G   \n",
       "1          SNP      NM_017641                       12  39709031   C   T   \n",
       "2          SNP      NM_017641                       12  39709057   C   A   \n",
       "3          SNP      NM_017641                       12  39735348   C   A   \n",
       "4          SNP      NM_017641                       12  39735937   G   C   \n",
       "\n",
       "             hgvs_g vcf_chrom  vcf_pos vcf_ref vcf_alt  \n",
       "0  12:g.39698647A>G       NaN      NaN     NaN     NaN  \n",
       "1  12:g.39709031C>T       NaN      NaN     NaN     NaN  \n",
       "2  12:g.39709057C>A       NaN      NaN     NaN     NaN  \n",
       "3  12:g.39735348C>A       NaN      NaN     NaN     NaN  \n",
       "4  12:g.39735937G>C       NaN      NaN     NaN     NaN  "
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants2['vcf_chrom'] = np.where(lovd_variants2['vcf_chrom'].isna(), lovd_variants2['Chromosome/scaffold name'], lovd_variants2['vcf_chrom'])\n",
    "lovd_variants2['vcf_pos'] = np.where(lovd_variants2['vcf_pos'].isna(), lovd_variants2['pos'], lovd_variants2['vcf_pos'])\n",
    "lovd_variants2['vcf_ref'] = np.where(lovd_variants2['vcf_ref'].isna(), lovd_variants2['ref'], lovd_variants2['vcf_ref'])\n",
    "lovd_variants2['vcf_alt'] = np.where(lovd_variants2['vcf_alt'].isna(), lovd_variants2['alt'], lovd_variants2['vcf_alt'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove unnecessary columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants2.drop(columns=['RefSeq mRNA ID','Chromosome/scaffold name','pos','ref','alt','hgvs_g'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DB_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>DNA_change_cDNA</th>\n",
       "      <th>DNA_change_genomic_hg19</th>\n",
       "      <th>Reported</th>\n",
       "      <th>Clinical_classification</th>\n",
       "      <th>Variant_remarks</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>vcf_chrom</th>\n",
       "      <th>vcf_pos</th>\n",
       "      <th>vcf_ref</th>\n",
       "      <th>vcf_alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIF21A_000001</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.4497T&gt;C</td>\n",
       "      <td>g.39698647A&gt;G</td>\n",
       "      <td>1</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>12</td>\n",
       "      <td>39698647</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KIF21A_000002</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3920G&gt;A</td>\n",
       "      <td>g.39709031C&gt;T</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>12</td>\n",
       "      <td>39709031</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KIF21A_000003</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.3909-15G&gt;T</td>\n",
       "      <td>g.39709057C&gt;A</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>12</td>\n",
       "      <td>39709057</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KIF21A_000004</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.1841G&gt;T</td>\n",
       "      <td>g.39735348C&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland, 174 he...</td>\n",
       "      <td>Faruq 2020, submtted</td>\n",
       "      <td>SNP</td>\n",
       "      <td>12</td>\n",
       "      <td>39735348</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KIF21A_000005</td>\n",
       "      <td>KIF21A</td>\n",
       "      <td>NM_017641.3</td>\n",
       "      <td>c.1674-11C&gt;G</td>\n",
       "      <td>g.39735937G&gt;C</td>\n",
       "      <td>1</td>\n",
       "      <td>likely benign</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "      <td>-</td>\n",
       "      <td>SNP</td>\n",
       "      <td>12</td>\n",
       "      <td>39735937</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           DB_ID    Gene   Transcript DNA_change_cDNA DNA_change_genomic_hg19  \\\n",
       "0  KIF21A_000001  KIF21A  NM_017641.3       c.4497T>C           g.39698647A>G   \n",
       "1  KIF21A_000002  KIF21A  NM_017641.3       c.3920G>A           g.39709031C>T   \n",
       "2  KIF21A_000003  KIF21A  NM_017641.3    c.3909-15G>T           g.39709057C>A   \n",
       "3  KIF21A_000004  KIF21A  NM_017641.3       c.1841G>T           g.39735348C>A   \n",
       "4  KIF21A_000005  KIF21A  NM_017641.3    c.1674-11C>G           g.39735937G>C   \n",
       "\n",
       "   Reported Clinical_classification  \\\n",
       "0         1                  benign   \n",
       "1         1           likely benign   \n",
       "2         1           likely benign   \n",
       "3         3                  benign   \n",
       "4         1           likely benign   \n",
       "\n",
       "                                     Variant_remarks             Reference  \\\n",
       "0             VKGL data sharing initiative Nederland                     -   \n",
       "1             VKGL data sharing initiative Nederland                     -   \n",
       "2             VKGL data sharing initiative Nederland                     -   \n",
       "3  VKGL data sharing initiative Nederland, 174 he...  Faruq 2020, submtted   \n",
       "4             VKGL data sharing initiative Nederland                     -   \n",
       "\n",
       "  Variant_Type vcf_chrom   vcf_pos vcf_ref vcf_alt  \n",
       "0          SNP        12  39698647       A       G  \n",
       "1          SNP        12  39709031       C       T  \n",
       "2          SNP        12  39709057       C       A  \n",
       "3          SNP        12  39735348       C       A  \n",
       "4          SNP        12  39735937       G       C  "
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [],
   "source": [
    "def valid_bases(bases: str) -> bool:\n",
    "    if re.match(\"^[ATCG]+$\", bases):\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants2['Valid_bases'] = lovd_variants2.apply(lambda x: 'Y' if valid_bases(str(x['vcf_ref'])) and valid_bases(str(x['vcf_alt'])) else 'N', axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Output data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "lovd_variants2.to_csv(\"/Users/himanshujoshi/Desktop/LOVD variants/all_LOVD_variants_with_VCF.tsv.gz\", sep=\"\\t\", compression='gzip', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DB_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>DNA_change_cDNA</th>\n",
       "      <th>DNA_change_genomic_hg19</th>\n",
       "      <th>Reported</th>\n",
       "      <th>Clinical_classification</th>\n",
       "      <th>Variant_remarks</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>vcf_chrom</th>\n",
       "      <th>vcf_pos</th>\n",
       "      <th>vcf_ref</th>\n",
       "      <th>vcf_alt</th>\n",
       "      <th>Valid_bases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>62757</th>\n",
       "      <td>ABCA4_001010</td>\n",
       "      <td>ABCA4</td>\n",
       "      <td>NM_000350.2</td>\n",
       "      <td>c.3583_3584insNN</td>\n",
       "      <td>g.94505622_94505623insNN</td>\n",
       "      <td>2</td>\n",
       "      <td>VUS</td>\n",
       "      <td>1 more item</td>\n",
       "      <td>PubMed: Cornelis 2017  ,  Journal: Cornelis 20...</td>\n",
       "      <td>Other</td>\n",
       "      <td>1</td>\n",
       "      <td>9.45056e+07</td>\n",
       "      <td>A</td>\n",
       "      <td>ANN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253284</th>\n",
       "      <td>TSC2_000213</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>NM_000548.3</td>\n",
       "      <td>c.3673_3674insNNNNN</td>\n",
       "      <td>g.2131658_2131659insNNNNN</td>\n",
       "      <td>1</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>5bp insertion (bases not specified); reported ...</td>\n",
       "      <td>PubMed: Jones, 1999</td>\n",
       "      <td>Other</td>\n",
       "      <td>16</td>\n",
       "      <td>2.13166e+06</td>\n",
       "      <td>A</td>\n",
       "      <td>ANNNNN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97311</th>\n",
       "      <td>AR_000514</td>\n",
       "      <td>AR</td>\n",
       "      <td>NM_000044.3</td>\n",
       "      <td>c.1998A&gt;Y</td>\n",
       "      <td>g.66931356A&gt;Y</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Steinkamp et al. Cancer Res 69: 4434-4442, 2009</td>\n",
       "      <td>Other</td>\n",
       "      <td>X</td>\n",
       "      <td>6.69314e+07</td>\n",
       "      <td>AA</td>\n",
       "      <td>&lt;Y&gt;</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62421</th>\n",
       "      <td>ABCA4_000663</td>\n",
       "      <td>ABCA4</td>\n",
       "      <td>NM_000350.2</td>\n",
       "      <td>c.3043T&gt;S</td>\n",
       "      <td>g.94510176A&gt;S</td>\n",
       "      <td>3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PubMed: Passerini 2010</td>\n",
       "      <td>Other</td>\n",
       "      <td>1</td>\n",
       "      <td>9.45102e+07</td>\n",
       "      <td>AA</td>\n",
       "      <td>&lt;S&gt;</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71236</th>\n",
       "      <td>IL1RAPL1_000017</td>\n",
       "      <td>IL1RAPL1</td>\n",
       "      <td>NM_014271.3</td>\n",
       "      <td>c.1854A&gt;Y</td>\n",
       "      <td>g.29973700A&gt;Y</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PubMed: Piton 2008  ,  Journal: Piton 2008</td>\n",
       "      <td>Other</td>\n",
       "      <td>X</td>\n",
       "      <td>2.99737e+07</td>\n",
       "      <td>AA</td>\n",
       "      <td>&lt;Y&gt;</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321462</th>\n",
       "      <td>KHDC3L_000005</td>\n",
       "      <td>KCNQ5</td>\n",
       "      <td>NM_019842.3</td>\n",
       "      <td>c.399-4566_*2910112del</td>\n",
       "      <td>g.73709065_76815249del</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>decreased gene dosage</td>\n",
       "      <td>PubMed: DDDS 2015  ,  Journal: DDDS 2015</td>\n",
       "      <td>Other</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321566</th>\n",
       "      <td>FAM161A_000017</td>\n",
       "      <td>FAM161A</td>\n",
       "      <td>NM_001201543.1</td>\n",
       "      <td>c.1355_1356del</td>\n",
       "      <td>g.62066781_62066782del, g.62066784_62066785del</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>1 more item</td>\n",
       "      <td>Sharon, submitted</td>\n",
       "      <td>Other</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321611</th>\n",
       "      <td>IMPDH1_000003</td>\n",
       "      <td>PODXL</td>\n",
       "      <td>NM_001018111.2</td>\n",
       "      <td>c.-4332841_*10189006del</td>\n",
       "      <td>g.121000064_135573959del</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>decreased gene dosage</td>\n",
       "      <td>PubMed: DDDS 2015  ,  Journal: DDDS 2015</td>\n",
       "      <td>Other</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321667</th>\n",
       "      <td>CASK_000037</td>\n",
       "      <td>MAOB</td>\n",
       "      <td>NM_000898.4</td>\n",
       "      <td>c.-326_*1980391del</td>\n",
       "      <td>g.41646322_43741871del</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>decreased gene dosage</td>\n",
       "      <td>PubMed: DDDS 2015  ,  Journal: DDDS 2015</td>\n",
       "      <td>Other</td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321975</th>\n",
       "      <td>DKC1_000000</td>\n",
       "      <td>BRCC3</td>\n",
       "      <td>NM_024332.3</td>\n",
       "      <td>c.-733962_*372950dup</td>\n",
       "      <td>g.153565841_154722429dup</td>\n",
       "      <td>1</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>increased gene dosage</td>\n",
       "      <td>PubMed: DDDS 2015  ,  Journal: DDDS 2015</td>\n",
       "      <td>Other</td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11909 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  DB_ID      Gene      Transcript          DNA_change_cDNA  \\\n",
       "62757      ABCA4_001010     ABCA4     NM_000350.2         c.3583_3584insNN   \n",
       "253284      TSC2_000213      TSC2     NM_000548.3      c.3673_3674insNNNNN   \n",
       "97311         AR_000514        AR     NM_000044.3                c.1998A>Y   \n",
       "62421      ABCA4_000663     ABCA4     NM_000350.2                c.3043T>S   \n",
       "71236   IL1RAPL1_000017  IL1RAPL1     NM_014271.3                c.1854A>Y   \n",
       "...                 ...       ...             ...                      ...   \n",
       "321462    KHDC3L_000005     KCNQ5     NM_019842.3   c.399-4566_*2910112del   \n",
       "321566   FAM161A_000017   FAM161A  NM_001201543.1           c.1355_1356del   \n",
       "321611    IMPDH1_000003     PODXL  NM_001018111.2  c.-4332841_*10189006del   \n",
       "321667      CASK_000037      MAOB     NM_000898.4       c.-326_*1980391del   \n",
       "321975      DKC1_000000     BRCC3     NM_024332.3     c.-733962_*372950dup   \n",
       "\n",
       "                               DNA_change_genomic_hg19  Reported  \\\n",
       "62757                         g.94505622_94505623insNN         2   \n",
       "253284                       g.2131658_2131659insNNNNN         1   \n",
       "97311                                    g.66931356A>Y         2   \n",
       "62421                                    g.94510176A>S         3   \n",
       "71236                                    g.29973700A>Y         2   \n",
       "...                                                ...       ...   \n",
       "321462                          g.73709065_76815249del         1   \n",
       "321566  g.62066781_62066782del, g.62066784_62066785del         4   \n",
       "321611                        g.121000064_135573959del         1   \n",
       "321667                          g.41646322_43741871del         1   \n",
       "321975                        g.153565841_154722429dup         1   \n",
       "\n",
       "       Clinical_classification  \\\n",
       "62757                      VUS   \n",
       "253284              pathogenic   \n",
       "97311                        -   \n",
       "62421                        -   \n",
       "71236                        -   \n",
       "...                        ...   \n",
       "321462                       -   \n",
       "321566                       -   \n",
       "321611                       -   \n",
       "321667                       -   \n",
       "321975              pathogenic   \n",
       "\n",
       "                                          Variant_remarks  \\\n",
       "62757                                         1 more item   \n",
       "253284  5bp insertion (bases not specified); reported ...   \n",
       "97311                                                   -   \n",
       "62421                                                   -   \n",
       "71236                                                   -   \n",
       "...                                                   ...   \n",
       "321462                              decreased gene dosage   \n",
       "321566                                        1 more item   \n",
       "321611                              decreased gene dosage   \n",
       "321667                              decreased gene dosage   \n",
       "321975                              increased gene dosage   \n",
       "\n",
       "                                                Reference Variant_Type  \\\n",
       "62757   PubMed: Cornelis 2017  ,  Journal: Cornelis 20...        Other   \n",
       "253284                                PubMed: Jones, 1999        Other   \n",
       "97311     Steinkamp et al. Cancer Res 69: 4434-4442, 2009        Other   \n",
       "62421                              PubMed: Passerini 2010        Other   \n",
       "71236          PubMed: Piton 2008  ,  Journal: Piton 2008        Other   \n",
       "...                                                   ...          ...   \n",
       "321462           PubMed: DDDS 2015  ,  Journal: DDDS 2015        Other   \n",
       "321566                                  Sharon, submitted        Other   \n",
       "321611           PubMed: DDDS 2015  ,  Journal: DDDS 2015        Other   \n",
       "321667           PubMed: DDDS 2015  ,  Journal: DDDS 2015        Other   \n",
       "321975           PubMed: DDDS 2015  ,  Journal: DDDS 2015        Other   \n",
       "\n",
       "       vcf_chrom      vcf_pos vcf_ref vcf_alt Valid_bases  \n",
       "62757          1  9.45056e+07       A     ANN           N  \n",
       "253284        16  2.13166e+06       A  ANNNNN           N  \n",
       "97311          X  6.69314e+07      AA     <Y>           N  \n",
       "62421          1  9.45102e+07      AA     <S>           N  \n",
       "71236          X  2.99737e+07      AA     <Y>           N  \n",
       "...          ...          ...     ...     ...         ...  \n",
       "321462         6          NaN     NaN     NaN           N  \n",
       "321566         2          NaN     NaN     NaN           N  \n",
       "321611         7          NaN     NaN     NaN           N  \n",
       "321667         X          NaN     NaN     NaN           N  \n",
       "321975         X          NaN     NaN     NaN           N  \n",
       "\n",
       "[11909 rows x 15 columns]"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants2.query('Valid_bases == \"N\"').sort_values(by='vcf_ref')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Valid_bases</th>\n",
       "      <th>N</th>\n",
       "      <th>Y</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variant_Type</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>11909</td>\n",
       "      <td>49841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP</th>\n",
       "      <td>0</td>\n",
       "      <td>260251</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Valid_bases       N       Y\n",
       "Variant_Type               \n",
       "Other         11909   49841\n",
       "SNP               0  260251"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.crosstab(lovd_variants2.Variant_Type, lovd_variants2.Valid_bases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>192374</th>\n",
       "      <th>266689</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DB_ID</th>\n",
       "      <td>TCAP_000010</td>\n",
       "      <td>SLCO1B3_000010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <td>TCAP</td>\n",
       "      <td>SLCO1B3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Transcript</th>\n",
       "      <td>NM_003673.3</td>\n",
       "      <td>NM_019844.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA_change_cDNA</th>\n",
       "      <td>c.37_39del</td>\n",
       "      <td>c.550G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA_change_genomic_hg19</th>\n",
       "      <td>g.37821649_37821651del</td>\n",
       "      <td>g.21015414G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reported</th>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Clinical_classification</th>\n",
       "      <td>VUS, benign, likely benign, pathogenic (recess...</td>\n",
       "      <td>VUS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variant_remarks</th>\n",
       "      <td>not in 400 control chromosomes, 0/144 DCM pati...</td>\n",
       "      <td>VKGL data sharing initiative Nederland</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reference</th>\n",
       "      <td>PubMed: Bos 2006  ,  PubMed: Marziliano 2006  ...</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variant_Type</th>\n",
       "      <td>Other</td>\n",
       "      <td>SNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vcf_chrom</th>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vcf_pos</th>\n",
       "      <td>3.78216e+07</td>\n",
       "      <td>21015414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vcf_ref</th>\n",
       "      <td>GGAG</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vcf_alt</th>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Valid_bases</th>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                    192374  \\\n",
       "DB_ID                                                          TCAP_000010   \n",
       "Gene                                                                  TCAP   \n",
       "Transcript                                                     NM_003673.3   \n",
       "DNA_change_cDNA                                                 c.37_39del   \n",
       "DNA_change_genomic_hg19                             g.37821649_37821651del   \n",
       "Reported                                                                16   \n",
       "Clinical_classification  VUS, benign, likely benign, pathogenic (recess...   \n",
       "Variant_remarks          not in 400 control chromosomes, 0/144 DCM pati...   \n",
       "Reference                PubMed: Bos 2006  ,  PubMed: Marziliano 2006  ...   \n",
       "Variant_Type                                                         Other   \n",
       "vcf_chrom                                                               17   \n",
       "vcf_pos                                                        3.78216e+07   \n",
       "vcf_ref                                                               GGAG   \n",
       "vcf_alt                                                                  G   \n",
       "Valid_bases                                                              Y   \n",
       "\n",
       "                                                         266689  \n",
       "DB_ID                                            SLCO1B3_000010  \n",
       "Gene                                                    SLCO1B3  \n",
       "Transcript                                          NM_019844.3  \n",
       "DNA_change_cDNA                                        c.550G>A  \n",
       "DNA_change_genomic_hg19                           g.21015414G>A  \n",
       "Reported                                                      1  \n",
       "Clinical_classification                                     VUS  \n",
       "Variant_remarks          VKGL data sharing initiative Nederland  \n",
       "Reference                                                     -  \n",
       "Variant_Type                                                SNP  \n",
       "vcf_chrom                                                    12  \n",
       "vcf_pos                                                21015414  \n",
       "vcf_ref                                                       G  \n",
       "vcf_alt                                                       A  \n",
       "Valid_bases                                                   Y  "
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lovd_variants2.sample(2).transpose()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
